src/main/java/core/basesyntax/fruitshop/impl/WriterServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.fruitshop.impl;
+
+import core.basesyntax.fruitshop.service.WriterService;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class WriterServiceImpl implements WriterService {
+    @Override
+    public void writeReport(Map<String, Integer> fruitInventory,
+                            String filePath) throws IOException {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            writer.write("fruit,quantity\n");
+            for (Map.Entry<String, Integer> entry : fruitInventory.entrySet()) {
+                writer.write(entry.getKey() + "," + entry.getValue() + "\n");
+            }
+        } catch (IOException e) {
+            throw new IOException("Error writing report to file", e);
```

Comment:
```suggestion
            throw new RuntimeException("Error writing report to file " + filePath, e);
```
always include filename to the exception messages related to files, fix in reader too